src/ts/components/seekbar.ts

```diff
@@ -607,30 +612,49 @@ export class SeekBar extends Component<SeekBarConfig> {
 
   protected updateMarkers(): void {
     this.seekBarMarkersContainer.empty();
+
+    const seekBarWidthPx = this.seekBar.width();
+
     for (let marker of this.timelineMarkers) {
+      const markerClasses = ['seekbar-marker'].concat(marker.marker.cssClasses || [])
+        .map(cssClass => this.prefixCss(cssClass));
+
+      const cssProperties: {[propertyName: string]: string} = {
+        'width': marker.position + '%',
+      };
+
+      if (marker.duration > 0) {
+        const markerWidthPx = Math.round(seekBarWidthPx / 100 * marker.duration);
+        cssProperties['border-right-width'] = markerWidthPx + 'px';
+        cssProperties['margin-left'] = '0';
+      }
+
       this.seekBarMarkersContainer.append(new DOM('div', {
-        'class': this.prefixCss('seekbar-marker'),
-        'data-marker-time': String(marker.time),
-        'data-marker-title': String(marker.title),
-      }).css({
-        'width': marker.time + '%',
-      }));
+        'class': markerClasses.join(' '),
+        'data-marker-time': String(marker.marker.time),
+        'data-marker-title': String(marker.marker.title),
+      }).css(cssProperties));
     }
   }
 
-  protected getMarkerAtPosition(percentage: number): TimelineMarker | null {
-    let snappedMarker: TimelineMarker = null;
-    let snappingRange = 1;
+  protected getMarkerAtPosition(percentage: number): SeekBarMarker | null {
+    const snappingRange = 1;
+
     if (this.timelineMarkers.length > 0) {
       for (let marker of this.timelineMarkers) {
-        if (percentage >= marker.time - snappingRange && percentage <= marker.time + snappingRange) {
-          snappedMarker = marker;
-          break;
+        if (marker.duration > 0
+          && percentage >= marker.position - snappingRange
+          && percentage <= marker.position + marker.duration + snappingRange) {
+          return marker;
+        }
+        else if (percentage >= marker.position - snappingRange
+          && percentage <= marker.position + snappingRange) {
+          return marker;
```

Comment:
I'm not a fan of multi-line if statements. How about something more expressive and clean?
```ts
if (this.timelineMarkers.length > 0) {
  for (let marker of this.timelineMarkers) {
    const isMarkerInStartRange = percentage >= marker.position - snappingRange;
    const isMarkerInEndRange = percentage <= marker.position + marker.duration + snappingRange;

    if (marker.duration > 0 && isMarkerInStartRange && isMarkerInEndRange) {
      return marker;
    } else if (isMarkerInStartRange && percentage <= marker.position + snappingRange) {
      return marker;
    }
  }
}
```

The variable names could of course be chosen better etc. And the example could be written better but you get my point 👍

Reply:
I wrote it on purpose the way it is because in your proposal the `isMarkerInEndRange` is always calculated although only needed in case `marker.duration > 0` 😝 But you're right, it's hard to follow.